enhancements/machine-api/dedicated-instances.md

```diff
@@ -0,0 +1,129 @@
+---
+title: dedicated-instances
+authors:
+  - "@alexander-demichev"
+reviewers:
+  - "@JoelSpeed"
+  - "@enxebre"
+approvers:
+  - "@JoelSpeed"
+  - "@enxebre"
+creation-date: 2020-09-01
+last-updated: 2020-09-01
+status: provisional
+---
+
+# Dedicated instances
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Make possible for users to create machines which run as dedicated instances. 
+
+## Motivation
+
+Some organizations need to make sure that their workloads are not hosted on the same physical hardware as others.
+
+### Goals
+
+- Provide automation similar to what Machine API supports for spot instances. 
+
+### Non-Goals
+
+- TODO
+
+## Proposal
+
+In order to give users ability to run their workloads on dedicated instances we should do the following things for AWS, GCP and Azure:
+
+- Add ability to enable dedicated instances using Machine's provider spec.
+
+- Validate that provider spec doesn't contain spot instances configuration and dedicated instances at the same time when it's not supported by the cloud provider.
+
+- Make sure that some required changes are supported and implemented on OCP installer side, like proper control plane creation.
+
+### Implementation Details
+
+For each of three cloud providers that support dedicated instances the implementation will be different.
+
+#### AWS
+
+`Dedicated Instances are Amazon EC2 instances that run in a virtual private cloud (VPC) on hardware that's dedicated to a single customer.`. [AWS Documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/dedicated-instance.html). Each launched instance has a tenancy attribute and it can be configured similar to how we set availability zone.
+
+```
+placement = &ec2.Placement{
+  AvailabilityZone: aws.String(machineProviderConfig.Placement.AvailabilityZone),
+  Tenancy: aws.String(machineProviderConfig.Placement.Tenancy)
+}
+```
+
+That change will require adding `Tenancy` field to provider spec.
```

Comment:
Might be good to see an example of the field as it would be implemented in Go, eg

```
type AWSMachineProviderConfig struct {
  // existing fields
  ...

  // Tenancy does some awesome thing
  Tenancy string
}
```

Filling out the comment on the field as if it were being properly documented.

Same goes for GCP and Azure too